libs/setup.sh

```diff
@@ -147,6 +148,7 @@ declare -A secondary_urls=(
   ["protobuf"]="https://github.com/protocolbuffers/protobuf.git"
   ["pulsar"]="https://github.com/apache/pulsar.git"
   ["librdtsc"]="https://github.com/gabrieleara/librdtsc.git"
+  ["ctre"]="https://raw.githubusercontent.com/hanickadot/compile-time-regular-expressions/v3.7.2/single-header/ctre.hpp"
```

Comment:
@Ignition does this compile on ARM? In general when adding libraries we have to make sure:
  * the library is well maintained -> this one seems to be (20+ contributors, recent activity, works on multiple compilers)
  * it works on ARM -> not sure
  * not too hard to build (header only or cmake setup) -> seems fine as well

Reply:
Should be fine for ARM. 
Its all pure C++ and also constexpr hence no UB.
Since it evaluates inside a static_assert I'm 100% certain it will be fine on all platforms.
https://godbolt.org/z/Gr7bohqad

Reply:
Also if you ever want to geek out, this is a good deep dive into the library https://youtu.be/IO3MO450WX4